pages/api/id/[form]/retrieval.ts

```diff
@@ -0,0 +1,207 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { logMessage } from "@lib/logger";
+import { middleware, cors, validTemporaryToken, jsonValidator } from "@lib/middleware";
+import retrievalSchema from "@lib/middleware/schemas/retrieval.schema.json";
+import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
+import {
+  DynamoDBDocumentClient,
+  QueryCommand,
+  UpdateCommand,
+  QueryCommandOutput,
+} from "@aws-sdk/lib-dynamodb";
+import { MiddlewareProps } from "@lib/types";
+
+interface ResponseListResult {
+  formResponses: QueryCommandOutput;
+  documentClient: DynamoDBDocumentClient;
+}
+
+/**
+ * Handler for the retrieval API route. This function simply calls the relevant function depending on the HTTP method
+ * @param req - The HTTP request object
+ * @param res - The HTTP response object
+ * @param email - The email of the user triggering the request
+ * @param temporaryToken - The JWT token of the user triggering the request
+ */
+const handler = async (
+  req: NextApiRequest,
+  res: NextApiResponse,
+  { email, temporaryToken }: MiddlewareProps
+) => {
+  const formID = req.query.form;
+  if (Array.isArray(formID) || !formID) return res.status(400).json({ error: "Bad Request" });
+  if (req.method === "DELETE") {
+    await deleteFormResponses(req, res, formID, email, temporaryToken);
+  } else {
+    await getFormResponses(req, res, formID, email, temporaryToken);
+  }
+};
+
+/**
+ * Helper function to instantiate DynamoDB and Document client.
+ * https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/dynamodb-example-document-client.html
+ */
+function connectToDynamo(): DynamoDBDocumentClient {
+  //Create dynamodb client
+  const db = new DynamoDBClient({
+    region: process.env.AWS_REGION ?? "ca-central-1",
+    endpoint: process.env.LOCAL_DYNAMODB_ENDPOINT,
```

Comment:
Great addition! Wondering if we should not be just adding one env variable named `LOCAL_AWS_ENDPOINT` since it is the same address we use for two different services already and https://github.com/cds-snc/platform-forms-client/pull/672 will benefit from it as well. Right now in my implementation I decided to use the existing `LOCAL_S3_ENDPOINT` (with a comment) even though it is for the SNS client.

Reply:
Yes definitely could probably be a single environment variable. I would have to modify code I'm not currently working on for this PR so I would prefer a separate one to do this

Reply:
I can definitely tackle this as part of my https://github.com/cds-snc/platform-forms-client/pull/672 once this is merged.